packages/teams-js/src/public/dialog.ts

```diff
@@ -17,21 +17,30 @@ import { runtime } from './runtime';
  *
  * @beta
  */
-
-export interface SdkResponse {
-  err?: string;
-  result?: string | object;
-}
-
 export namespace dialog {
+  /**
+   * Data Structure to represent the SDK response when dialog closes
+   */
+  export interface ISdkResponse {
+    /**
+     * Error in case there is a failure before dialog submission
+     */
+    err?: string;
+
+    /**
+     * Result value that the dialog is submitted with
+     */
+    result?: string | object;
+  }
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   export type PostMessageChannel = (message: any) => void;
-  export type DialogSubmitHandler = (result: SdkResponse) => void;
+  export type DialogSubmitHandler = (result: ISdkResponse) => void;
+
   /**
    * Allows an app to open the dialog module.
    *
    * @param urlDialogInfo - An object containing the parameters of the dialog module.
-   * @param submitHandler - This Handler is called when the dialog has been submitted or closed.
+   * @param submitHandler - Handler that triggers when the dialog has been submitted or closed.
```

Comment:
Probably worth again linking to the submit function and saying that this function will be called when a dialog calls the submit function OR when the user closes the dialog.